package/MDAnalysis/analysis/align.py

```diff
@@ -682,19 +686,24 @@ def rms_fit_trj(
         logging.disable(logging.WARN)
 
     kwargs.setdefault('remarks', 'RMS fitted trajectory to reference')
-    if filename is None:
-        path, fn = os.path.split(frames.filename)
-        filename = os.path.join(path, prefix + fn)
-        _Writer = frames.Writer
+    writer = None
+    if in_memory:
+        traj.transfer_to_memory()
+        frames = traj.trajectory
```

Comment:
Do you mean moving the `frames = traj.trajectory` line from the first line of the method (line 684) to after the if-block (line 709). If so, this would not work, since the else branch in the if-statement also uses the `frames` variable. Perhaps I'm misunderstanding what you meant.